tests/unittests/AddressBook/BasicQueryTests.mm

```diff
@@ -0,0 +1,115 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <TestFramework.h>
+
+#import <CoreFoundation/CoreFoundation.h>
+#import <Foundation/Foundation.h>
+
+#import "UWP/WindowsApplicationModelContacts.h"
+#import <AddressBook/ABAddressBook.h>
+#import <AddressBook/ABRecord.h>
+#import <AddressBook/ABPerson.h>
+#import "ABContactInternal.h"
+
+class AddressBookQueryTest : public ::testing::TestWithParam<::testing::tuple<ABPropertyID, NSString*>> {
+protected:
+    virtual void SetUp() {
+        WACContact* contact = [WACContact make];
+        contact.notes = @"Test Note!";
+        contact.nickname = @"Mikey";
+        contact.yomiGivenName = @"Mic";
+        contact.honorificNameSuffix = @"Jr.";
+        contact.yomiFamilyName = @"Saw-ft";
+        contact.middleName = @"Roe";
+        contact.lastName = @"Soft";
+        contact.honorificNamePrefix = @"Mr.";
+        contact.firstName = @"Mike";
+
+        WACContactJobInfo* job = [WACContactJobInfo make];
+        job.title = @"SDE";
+        job.office = @"Redmond";
+        job.manager = @"Satya";
+        job.Description = @"I develop good code.";
+        job.department = @"WDG";
+        job.companyYomiName = @"Mike Roe Soft";
+        job.companyName = @"Microsoft";
+        job.companyAddress = @"Redmond, WA";
+
+        [contact.jobInfo addObject:job];
+        _record = (__bridge_retained ABRecordRef)[[_ABContact alloc] initWithContact:contact];
+    }
+
+    virtual void TearDown() {
+        CFRelease(_record);
+    }
+
+    ABRecordRef _record;
+};
+
+TEST_P(AddressBookQueryTest, BasicContactQuery) {
+    ABPropertyID property = ::testing::get<0>(GetParam());
+    ASSERT_OBJCEQ_MSG(::testing::get<1>(GetParam()),
+                      (__bridge_transfer NSString*)ABRecordCopyValue(_record, property),
```

Comment:
All of the ones tested in this case are Strings -- the only current non-string value is a Date, which is tested in the birthday test. 

---
In reply to: [74829896](https://github.com/Microsoft/WinObjC/pull/881#discussion_r74829896) [](ancestors = 74829896)